### PR TITLE
DOC: overhaul

### PR DESCRIPTION
**Pull request by danielballan**
- Remove usage examples that are more focused on DataMuxer usage.
- Add simple examples to index page.
- Split "API docs" into more obviously-titled "Searching for Data" and "Fetching Data".
- Add section explaining scan ID vs. uid (yesterday I learned that even CSX is still confused about this, so we docs on this we can point people to).

### CODE REVIEW

**Comment by danielballan** [comment]
Built docs are already pushed [here](http://nsls-ii.github.io/databroker/)

File: doc/source/index.rst [hunk 1]
```diff
@@ -13,11 +13,44 @@ built-in Python data types and numpy arrays.
...
+
+If they are not already imported, import ``DataBroker`` and its convenience
+fucntions. (This step may already be done as part of an
+`IPython profile <http://nsls-ii.github.io/beamline-configuration.html#ipython-profiles>`_.)
+
+.. code-block:: python
+
+    from databroker import DataBroker as db, get_table, get_images, get_events
+
+Example 1: Load scalar data in a table.
```
Comment (ericdill) [inline doc/source/index.rst]: I appreciate the simplicity of the example.  Do you want to mention the `fill` kwarg which defaults to filling in external data? Maybe change the example to `get_table(last_run, fill=False)`?

**Comment by ericdill** [comment]
Just a minor quibble about the `scalar` language in the first example.  Otherwise LGTM!

**Comment by danielballan** [comment]
Good suggestion, @ericdill. Revised and self-merged.

**Pull request merged**